pkg/tasker/tasker.go

```diff
@@ -0,0 +1,179 @@
+		return nil, err
+	}
+	return &tellor.ITellorNewChallenge{
+		CurrentChallenge: newVariables.Challenge,
+		Difficulty:       newVariables.Difficutly,
+		CurrentRequestId: newVariables.RequestIds,
+		TotalTips:        newVariables.Tip,
+	}, err
+}
+
+func (mt *Tasker) getNewChallengeChannel() (chan *tellor.ITellorNewChallenge, event.Subscription, error) {
+	sink := make(chan *tellor.ITellorNewChallenge)
+	var tellorFilterer *tellor.ITellorFilterer
+	tellorFilterer, err := tellor.NewITellorFilterer(mt.contractInstance.Address, mt.client)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "getting ITellorFilterer instance")
+	}
+	sub, err := tellorFilterer.WatchNewChallenge(&bind.WatchOpts{}, sink, nil)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "getting NewChallenge channel")
+	}
+	return sink, sub, nil
+}
+
+func (mt *Tasker) sendWork(challenge *tellor.ITellorNewChallenge) {
+	if challenge.CurrentRequestId[0].Int64() > int64(100) || challenge.CurrentRequestId[0].Int64() == 0 {
+		level.Warn(mt.logger).Log("msg", "new current variables request ID not correct - contract about to be upgraded")
+		return
+	}
+	newChallenge := &mining.MiningChallenge{
+		Challenge:  challenge.CurrentChallenge[:],
+		Difficulty: challenge.Difficulty,
+		RequestIDs: challenge.CurrentRequestId,
+	}
+	for _, acc := range mt.accounts {
+		level.Info(mt.logger).Log("msg", "new challenge",
+			"addr", acc.Address.String(),
+			"challenge", fmt.Sprintf("%x", newChallenge.Challenge),
+			"difficulty", newChallenge.Difficulty,
+			"requestIDs", fmt.Sprintf("%+v", newChallenge.RequestIDs),
+		)
+		go func(acc *rpc.Account) {
+			mt.workSinks[acc.Address.String()] <- &mining.Work{Challenge: newChallenge, PublicAddr: acc.Address.String(), Start: uint64(rand.Int63()), N: math.MaxInt64}
+		}(acc)
+	}
+
+}
+
+func (mt *Tasker) Start() error {
+	var err error
+	level.Info(mt.logger).Log("msg", "tasker has been started")
+	currentChallenge, err := mt.getCurrentChallenge()
+
+	if err != nil {
+		return errors.Wrap(err, "tasker getting the current challenge")
+	}
+
+	level.Info(mt.logger).Log("msg", "tasker is sending the initial challenge to the miner")
+	mt.sendWork(currentChallenge)
+
+	// Subscribe and wait until the context cancellation event.
+	var sink chan *tellor.ITellorNewChallenge
+	var sub event.Subscription
+
+	// Initial subscription.
+	for {
+		sink, sub, err = mt.getNewChallengeChannel()
+		if err != nil {
+			level.Error(mt.logger).Log("msg", "initial subscribing to NewChallenge events failed")
+			time.Sleep(1 * time.Second)
+			continue
+		}
+		break
+	}
+
+	for {
+		select {
+		case <-mt.ctx.Done():
+			level.Info(mt.logger).Log("msg", "unsubscribing from NewChallenge events")
+			if sub != nil {
+				sub.Unsubscribe()
+			}
+			level.Info(mt.logger).Log("msg", "closing the tasker channels")
+			for _, ch := range mt.workSinks {
+				close(ch)
+			}
+			level.Info(mt.logger).Log("msg", "tasker shutdown complete")
```

Comment:
Lets put all shutdown messages in cmds.go in the run groups.
Similiar to what you have done for the metrics server. Just do the same for all groups. This is more consistent and will still give us an idea when some group blocks the shutdown.
```
level.Info(logger).Log("msg", "metrics server is closing")
```
```suggestion
```